pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -158,15 +156,14 @@ func mergeRules(originalRules, latestRules []*Rule) []*Rule {
 
 // mergeRoutesWeightedClusters merges two slices of RouteWeightedClusters and returns a slice where there is one RouteWeightedCluster
 //	for any HTTPRouteMatch
-func mergeRoutesWeightedClusters(originalRoutes, latestRoutes []*RouteWeightedClusters) ([]*RouteWeightedClusters, []error) {
-	mergeErrors := []error{}
+func mergeRoutesWeightedClusters(originalRoutes, latestRoutes []*RouteWeightedClusters) []*RouteWeightedClusters {
 	for _, latest := range latestRoutes {
 		foundRoute := false
 		for _, original := range originalRoutes {
 			if reflect.DeepEqual(original.HTTPRouteMatch, latest.HTTPRouteMatch) {
 				foundRoute = true
 				if !reflect.DeepEqual(original.WeightedClusters, latest.WeightedClusters) {
-					mergeErrors = append(mergeErrors, errors.Errorf("Error merging RoutesWeightedClusters %v with %v", original, latest))
+					original.WeightedClusters = latest.WeightedClusters
```

Comment:
Why do you take the latest and not take the union of both sets? Can you add a comment here?

Reply:
Each route match can only have one set of weighted clusters. 
Ex. 
**original.WeightedClusters**: {cluster1: 50, cluster2: 50}
**latest.WeightedClusters**: {cluster1: 20, cluster2: 80}

If you look at it like that, I think it makes more sense why we can't take the union. A route match is associated with an action. The action indicates the behavior on a route match. The original behavior (with the given example) is that each cluster gets 50 % of the traffic. In latest, cluster 1 gets 20% and cluster 2 gets 80%. When we merge RoutesWeightedClusters, we want to take the latest and replace the original. In practice, we first calculate outbound 
traffic policies using Traffic Targets. We can then separately calculate policies based on traffic split. When we merge the outbound policies, we want to ensure the traffic split related policies take precedence over traffic target if there is overlap.